File: outputs/verification/pp-education/check_console.py
#!/usr/bin/env python3
"""Check console errors."""
import os
from playwright.sync_api import sync_playwright

OUT = "/root/projects/brightkidco/outputs/verification/pp-education"
HTML = f"file://{OUT}/pped-11-d7-render.html"

errors = []
with sync_playwright() as p:
    browser = p.chromium.launch(args=["--no-sandbox", "--disable-gpu", "--disable-dev-shm-usage"])
    page = browser.new_page(viewport={"width": 420, "height": 900})
    page.on("console", lambda msg: errors.append(f"[{msg.type}] {msg.text}") if msg.type in ("error", "warning") else None)
    page.goto(HTML, wait_until="networkidle", timeout=30000)
    page.wait_for_timeout(2000)
    browser.close()

with open(f"{OUT}/console-errors.txt", "w") as f:
    if errors:
        f.write("\n".join(errors))
    else:
        f.write("No console errors or warnings\n")
print(f"Found {len(errors)} console issues")
for e in errors:
    print(e)

File: outputs/verification/pp-education/check_console.js
const puppeteer = require('/root/.hermes/node/lib/node_modules/puppeteer-core');
(async () => {
  const browser = await puppeteer.launch({
    executablePath: '/root/.cache/puppeteer/chrome/linux-149.0.7827.22/chrome-linux64/chrome',
    args: ['--no-sandbox', '--disable-gpu'],
    headless: 'new',
  });
  const page = await browser.newPage();
  const errors = [];
  page.on('console', msg => {
    if (msg.type() === 'error') errors.push(msg.text());
  });
  page.on('pageerror', err => errors.push(err.message));
  await page.setViewport({ width: 420, height: 800 });
  await page.goto('file:///root/projects/brightkidco/outputs/verification/pp-education/pped-11-d14-render.html', { waitUntil: 'networkidle0', timeout: 30000 });
  await new Promise(r => setTimeout(r, 2000));
  
  // Get page height
  const height = await page.evaluate(() => document.getElementById('root').offsetHeight);
  console.log('Page height:', height);
  
  if (errors.length === 0) {
    console.log('NO CONSOLE ERRORS');
  } else {
    console.log('CONSOLE ERRORS:');
    errors.forEach(e => console.log('  -', e));
  }
  await browser.close();
})();
